src/main/java/com/yelp/nrtsearch/server/utils/ThreadPoolExecutorFactory.java

```diff
@@ -36,19 +34,57 @@ public enum ExecutorType {
     REPLICATIONSERVER,
     FETCH,
     GRPC,
-    METRICS
+    METRICS,
+    VECTOR_MERGE
   }
 
   private static final Logger logger = LoggerFactory.getLogger(ThreadPoolExecutorFactory.class);
   private static final int DEFAULT_QUEUE_SIZE = 8;
 
+  private static ThreadPoolExecutorFactory instance;
+
+  private final ThreadPoolConfiguration threadPoolConfiguration;
+  private final Map<ExecutorType, ThreadPoolExecutor> executorMap = new ConcurrentHashMap<>();
+
+  /**
+   * Initialize the factory with the provided {@link ThreadPoolConfiguration}.
+   *
+   * @param threadPoolConfiguration thread pool configuration
+   */
+  public static void init(ThreadPoolConfiguration threadPoolConfiguration) {
+    instance = new ThreadPoolExecutorFactory(threadPoolConfiguration);
+  }
+
   /**
+   * Get the instance of the factory.
+   *
+   * @return instance of the factory
+   */
+  public static ThreadPoolExecutorFactory getInstance() {
+    return instance;
```

Comment:
Do you want to add a null check here and throw an exception if `instance` is not initialized? Would also require setting it to null initially.